### PR TITLE
tools: provide tool to create doxygen maintainer list

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This has been long discussed, but never set into action. Given that [MAINTAINING.md](https://github.com/RIOT-OS/RIOT/blob/master/MAINTAINING.md) now references a non-existent wiki page (which was outdated before its deletion), I say it is about time.

This basically takes the information from [the RIOT-OS org teams](https://github.com/orgs/RIOT-OS/teams) and combines it with the information in [CODEOWNERS](https://github.com/RIOT-OS/RIOT/blob/master/CODEOWNERS). Caveat: One needs a token with read:org permission to get the teams.

I tried to match the CODEOWNERS patterns to doxygen groups somehow, but there either too little overlap (only @jia200x who chose documentation as an area of expertise had matching doxygen groups) or too much (e.g., @kaspar030 then received nearly every CPU, because he is interested in `pm.c` which have an `@ingroup` for each respective CPU or their `periph` interface...), so I just gave up on that.

### Testing procedure
If all goes well, Murdock should just have a new `maintainers-list.html` in its Documentation preview.

### Issues/PRs references
None so far.

### CODE REVIEW

**Comment by miri64** [comment]
> ### Testing procedure
> 
> If all goes well, Murdock should just have a new `maintainers-list.html` in its Documentation preview.

That did not work. :-( @RIOT-OS/ci can you have a look?

**Comment by kaspar030** [comment]
I don't know how that new html is created or where it ends up, but these lines deal with the preview: https://github.com/RIOT-OS/murdock-scripts/blob/931a4136b6efa7d44539826d665ae2b1561e16e1/build.sh#L281

File: MAINTAINING.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ In case of security relevant backports (both bug fixes and reverts), the
 announcement can be skipped and the fix merged once at least two ACKs are
 there.
 
-[list of maintainers]: https://github.com/RIOT-OS/RIOT/wiki/Maintainers
+[list of maintainers]: https://doc.riot-os.org/maintainer-list.html
```
[the remark this answers is not in the document]
Comment (miri64) [reply MAINTAINING.md:184]: The PR is not merged yet and the tooling does not work, so no surprise there. :-)

[the remark this answers is not in the document]
Comment (jkarinkl) [reply MAINTAINING.md:184]: You may resolve this comment :).

**Comment by miri64** [comment]
https://github.com/RIOT-OS/RIOT/actions/runs/12201211174/job/34039104721 is failing for this PR since the build was initiated from my fork (where the secret is not present). However, I did some testing on a feature branch. There it worked: https://github.com/RIOT-OS/RIOT/actions/runs/12200980590

**Comment by miri64** [comment]
Provided https://github.com/RIOT-OS/riot-os.org/pull/141 which might be a better alternative

**Comment by miri64** [comment]
Closing in favor of https://github.com/RIOT-OS/riot-os.org/pull/141

**Pull request closed without merging**